hewr/R/process_state_forecast.R

```diff
@@ -30,135 +113,119 @@ process_state_forecast <- function(model_run_dir,
   )
   posterior_predictive <- arrow::read_parquet(posterior_predictive_path)
 
-
   other_ed_visits_path <- fs::path(timeseries_model_dir,
     "other_ed_visits_forecast",
     ext = "parquet"
   )
   other_ed_visits_forecast <- arrow::read_parquet(other_ed_visits_path) |>
-    dplyr::rename(Other = other_ed_visits)
+    dplyr::rename(Other = "other_ed_visits")
 
-  combined_dat <-
-    dplyr::bind_rows(
-      train_dat |>
-        dplyr::filter(data_type == "train"),
-      eval_dat
-    ) |>
-    dplyr::mutate(
-      disease = dplyr::if_else(
-        disease == disease_name_nssp,
-        "Disease", # assign a common name for
-        # use in plotting functions
-        disease
-      )
-    ) |>
-    tidyr::pivot_wider(names_from = disease, values_from = ed_visits) |>
-    dplyr::mutate(
-      Other = Total - Disease,
-      prop_disease_ed_visits = Disease / Total
-    ) |>
-    dplyr::select(-Total) |>
-    dplyr::mutate(time = dplyr::dense_rank(date)) |>
-    tidyr::pivot_longer(c(Disease, Other, prop_disease_ed_visits),
-      names_to = "disease",
-      values_to = ".value"
-    )
+  combined_dat <- combine_training_and_eval_data(
+    train_dat,
+    eval_dat,
+    disease_name
+  )
 
   last_training_date <- combined_dat |>
-    dplyr::filter(data_type == "train") |>
-    dplyr::pull(date) |>
+    dplyr::filter(.data$data_type == "train") |>
+    dplyr::pull("date") |>
     max()
 
   other_ed_visits_samples <-
```

Comment:
I know I wrote this, but I think this part could use a comment to explain that we are creating a fake posterior by replicating the true data.

Reply:
done in e1ff3d7